### PR TITLE
Drone starlark

### PR DESCRIPTION
**Pull request by phil-davis**
Issue #282 

- add code style, phan, phpstan tools
- move `phpunit.xml` up to the root folder, like in other repos
- implement code style changes
- implement drone starlark 

See PR https://github.com/owncloud/activity/pull/802 to see the diffs of what starlark code I changed to support this app.

### CODE REVIEW

File: .drone.starlark [hunk 1]
```diff
@@ -0,0 +1,1326 @@
...
+					]
+				}
+			],
+		},
+	},
+
+	'acceptance': False,
+}
+
+def main(ctx):
```
Comment (phil-davis) [inline .drone.starlark]: The starlark code from here is exactly the same as the code in `activity` app that has already been reviewed.

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,15 @@
-vendor
-build
-tests/output
+/build/
```
Comment (phil-davis) [inline .gitignore]: This is "standard" stufff that goes in gitignore, like other apps.

File: .phan/config.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
+<?php
```
Comment (phil-davis) [inline .phan/config.php]: This is the standaqrd phan config like other apps

File: .php_cs.dist [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?php
```
Comment (phil-davis) [inline .php_cs.dist]: This is a standard config, like other apps.

File: Makefile [hunk 5]
```diff
@@ -19,7 +19,6 @@ acceptance_test_deps=vendor-bin/behat/vendor
 PHPUNIT=php -d zend.enable_gc=0 ../../lib/composer/bin/phpunit
 PHPUNITDBG=phpdbg -qrr -d memory_limit=4096M -d zend.enable_gc=0 "../../lib/composer/bin/phpunit"
 PHP_CS_FIXER=php -d zend.enable_gc=0 vendor-bin/owncloud-codestyle/vendor/bin/php-cs-fixer
-PHP_CODESNIFFER=vendor-bin/php_codesniffer/vendor/bin/phpcs
```
Comment (phil-davis) [inline Makefile]: codesniffer is not used in this app, because there are no acceptance tests, and we only use codesnifffer for acceptance test code.

File: composer.json [hunk 6]
```diff
@@ -4,12 +4,23 @@
...
+            "php": "7.0.8"
+        }
     },
     "replace": {
         "guzzlehttp/guzzle": "*"
     },
     "require": {
         "aws/aws-sdk-php": "^3.93"
+    },
+    "require-dev": {
```
Comment (phil-davis) [inline composer.json]: This is the "standard" way that we do the `vendor-bin` tools, like in other apps.

File: phpstan.neon [hunk 7]
```diff
@@ -0,0 +1,4 @@
+parameters:
```
Comment (phil-davis) [inline phpstan.neon]: The is the "standard" phpstan config

File: phpunit.xml [hunk 8]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (phil-davis) [inline phpunit.xml]: phpunit has been moved to here and made similar to other apps, so that the phpunit CI works like other apps

File: tests/drone/configs/config.scality.php [hunk 9]
```diff
@@ -1,12 +1,12 @@
...
-    'port'          => '8000',
-    'key'           => 'owncloud123456',
-    'secret'        => 'secret123456',
-    'use_ssl'       => false,
-    'use_path_style'=> true,
-);
\ No newline at end of file
+return [
+	'run'           => true,
+	'bucket'        => 'owncloud',
```
Comment (skshetry) [inline tests/drone/configs/config.scality.php]: Why was this changed?

Comment (phil-davis) [reply tests/drone/configs/config.scality.php]: Other apps were already using `owncloud` as the name of the bucket in tests, so that was already in the starlark, to have a bucket called `owncloud`
So it is easy to use the "standard" from other app tests.
